java-checks/src/main/java/org/sonar/java/checks/StaticMembersAccessCheck.java

```diff
@@ -19,35 +19,75 @@
+import java.util.List;
 import org.sonar.check.Rule;
+import org.sonar.java.checks.helpers.QuickFixHelper;
+import org.sonar.java.reporting.JavaQuickFix;
+import org.sonar.java.reporting.JavaTextEdit;
 import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.JavaFileScannerContext;
+import org.sonar.plugins.java.api.semantic.Symbol;
+import org.sonar.plugins.java.api.semantic.Type;
 import org.sonar.plugins.java.api.tree.ExpressionTree;
 import org.sonar.plugins.java.api.tree.IdentifierTree;
 import org.sonar.plugins.java.api.tree.MemberSelectExpressionTree;
 import org.sonar.plugins.java.api.tree.Tree;
 
-import java.util.Collections;
-import java.util.List;
-
 @Rule(key = "S2209")
 public class StaticMembersAccessCheck extends IssuableSubscriptionVisitor {
 
+  private QuickFixHelper.ImportSupplier importSupplier;
+
   @Override
   public List<Tree.Kind> nodesToVisit() {
     return Collections.singletonList(Tree.Kind.MEMBER_SELECT);
   }
 
+  @Override
+  public void setContext(JavaFileScannerContext context) {
+    super.setContext(context);
+    importSupplier = null;
+  }
+
+  @Override
+  public void leaveFile(JavaFileScannerContext context) {
+    importSupplier = null;
+  }
+
   @Override
   public void visitNode(Tree tree) {
     MemberSelectExpressionTree memberSelect = (MemberSelectExpressionTree) tree;
-    if (memberSelect.identifier().symbol().isStatic()) {
-      ExpressionTree memberSelectExpression = memberSelect.expression();
-      if (memberSelectExpression.is(Tree.Kind.MEMBER_SELECT)) {
-        memberSelectExpression = ((MemberSelectExpressionTree) memberSelectExpression).identifier();
-      }
-      if (!memberSelectExpression.is(Tree.Kind.IDENTIFIER) || ((IdentifierTree) memberSelectExpression).symbol().isVariableSymbol()) {
-        context.reportIssue(this, memberSelect, "Change this instance-reference to a static reference.");
+    IdentifierTree memberSelectIdentifier = memberSelect.identifier();
+    Symbol memberSelectSymbol = memberSelectIdentifier.symbol();
+    if (memberSelectSymbol.isStatic()) {
+      ExpressionTree leftOperand = memberSelect.expression();
+      ExpressionTree selectExpression = leftOperand.is(Tree.Kind.MEMBER_SELECT)
+        ? ((MemberSelectExpressionTree) leftOperand).identifier()
+        : leftOperand;
+      if (!selectExpression.is(Tree.Kind.IDENTIFIER) || ((IdentifierTree) selectExpression).symbol().isVariableSymbol()) {
+        QuickFixHelper.newIssue(context)
+          .forRule(this)
+          .onTree(memberSelect)
+          .withMessage("Change this instance-reference to a static reference.")
+          .withQuickFix(() -> createQuickFixes(leftOperand, memberSelectSymbol.owner().type()))
+          .report();
       }
     }
   }
+
+  private JavaQuickFix createQuickFixes(ExpressionTree leftOperand, Type type) {
+    String leftOperandAsText = QuickFixHelper.contentForTree(leftOperand, context).replaceAll("\\s+", " ");
```

Comment:
I think this should be done in a simpler way. If it's an identifier (or a member select) we just use its name, if not, we simply mark "this expression"